rest-client/client-common/src/main/java/io/apicurio/registry/rest/client/impl/RegistryClientImpl.java

```diff
@@ -252,85 +271,97 @@ public ArtifactSearchResults searchArtifacts(String group, String name, String
         if (properties != null && !properties.isEmpty()) {
             queryParams.put(Parameters.PROPERTIES, properties);
         }
-        return requestHandler.sendRequest(SearchRequestsProvider.searchArtifacts(queryParams));
+        return registryHttpClient.sendRequest(SearchRequestsProvider.searchArtifacts(queryParams));
     }
 
     @Override
     public ArtifactSearchResults searchArtifactsByContent(InputStream data, SortBy orderBy, SortOrder order,
                                                           Integer offset, Integer limit) {
         final Map<String, List<String>> queryParams = new HashMap<>();
         checkCommonQueryParams(orderBy, order, limit, offset, queryParams);
-        return requestHandler.sendRequest(SearchRequestsProvider.searchArtifactsByContent(data, queryParams));
+        return registryHttpClient.sendRequest(SearchRequestsProvider.searchArtifactsByContent(data, queryParams));
     }
 
     @Override
     public List<RuleType> listGlobalRules() {
-        return requestHandler.sendRequest(AdminRequestsProvider.listGlobalRules());
+        return registryHttpClient.sendRequest(AdminRequestsProvider.listGlobalRules());
     }
 
     @Override
     public void createGlobalRule(Rule data) {
-        requestHandler.sendRequest(AdminRequestsProvider.createGlobalRule(data));
+        try {
+            registryHttpClient.sendRequest(AdminRequestsProvider.createGlobalRule(data));
+        } catch (JsonProcessingException e) {
+            throw parseSerializationError(e);
+        }
```

Comment:
I wonder why sometimes, you wrap the `sendRequest` call into `try`/`catch` to handle `JsonProcessingException`, and sometimes, you don't :-)

Reply:
I'm dealing with the exception being thrown inside, in this case, the AdminRequestProvider.createGlobalRule, that's the reason behind this.